solc/CommandLineInterface.cpp

```diff
@@ -1097,6 +1097,26 @@ void CommandLineInterface::outputCompilationResults()
 {
 	solAssert(m_options.input.mode == InputMode::Compiler || m_options.input.mode == InputMode::CompilerWithASTImport, "");
 
+	if (!m_options.compiler.outputs.signatureHashes &&
+		!m_options.compiler.outputs.ir &&
+		!m_options.compiler.outputs.irOptimized &&
+		!m_options.compiler.outputs.astCompactJson &&
+		!m_options.modelChecker.initialize &&
```

Comment:
I think that it should still be possible to enter this `if` when using model checker, `--hashes` or IR output. You can still request for example `--bin` together with all these and it can result in the compiler telling you `Compiler run successful, no output requested.`, which is not true.

 Consider this example:
```solidity
// SPDX-License-Identifier: GPL-2.0
pragma solidity >=0.0;

function f() pure {
    assert(true == true);
}
```
```bash
solc test.sol --model-checker-engine all --bin
```

Reply:
Alright, I will let them get there and update the expected outputs in cmdlineTests.

Reply:
Actually, please see https://github.com/ethereum/solidity/pull/12378#issuecomment-991356098 first.